tycho-integration-test/src/main.rs

```diff
@@ -519,95 +583,137 @@ async fn process_state(
-                        Some(block),
-                        Address::from_slice(&solution.sender[..20]),
-                    );
-                    // Generate overwrites string with metadata
-                    let overwrites_string = if let Some(overwrites) = state_overwrites.as_ref() {
-                        tenderly::get_overwites_string(overwrites, metadata.as_ref())
-                    } else {
-                        String::new()
-                    };
-
-                    error!(
-                        event_type = "simulation_execution_failure",
-                        error_message = %revert_reason,
-                        error_name = %error_name,
-                        tenderly_url = %tenderly_url,
-                        overwrites = %overwrites_string,
-                        "Failed to simulate swap: {error_msg}"
-                    );
-                    metrics::record_simulation_execution_failure(
-                        &component.protocol_system,
-                        &error_name,
-                    );
-
-                    continue;
-                }
-            };
-        info!("Simulated amount_out: {simulated_amount_out} {}", token_out.symbol);
-
-        // Calculate slippage: positive = simulated > expected, negative = simulated < expected
-        let slippage = if simulated_amount_out >= expected_amount_out {
-            let diff = &simulated_amount_out - &expected_amount_out;
-            ((diff.clone() * BigUint::from(10000u32)) / expected_amount_out)
-                .to_f64()
-                .unwrap_or(0.0) /
-                100.0
-        } else {
-            let diff = &expected_amount_out - &simulated_amount_out;
-            -((diff.clone() * BigUint::from(10000u32)) / expected_amount_out)
-                .to_f64()
-                .unwrap_or(0.0) /
-                100.0
-        };
-
-        info!(
-            event_type = "execution_slippage",
-            slippage_ratio = slippage,
-            "Execution slippage: {:.2}%",
-            slippage
+        execution_infos.insert(
+            format!("{}-{:?}", simulation_id, i),
+            TychoExecutionInfo {
+                solution,
+                transaction,
+                expected_amount_out,
+                protocol_system: component.protocol_system.clone(),
+                component_id: component.id.to_string(),
+            },
         );
-        metrics::record_execution_slippage(&component.protocol_system, slippage);
 
         info!(
             "{} pool processed {state_id} from {} to {}",
             component.protocol_system, token_in.symbol, token_out.symbol
         );
     }
+    execution_infos
+}
+
+#[tracing::instrument(
+    skip_all,
+    fields(
+        simulation_id = %simulation_id,
+        protocol = %execution_info.protocol_system,
+        block_number = %block.header.number,
+        component_id = %execution_info.component_id,
+    )
+)]
+fn process_execution_result(
```

Comment:
Smol docstring? I understand it sends some metrics and returns fail/success counts